Sources/LinuxPlatform/Linux.swift

```diff
@@ -141,5 +141,76 @@ public struct Linux: Platform {
         FileManager.default.temporaryDirectory.appendingPathComponent("swiftly-\(UUID())")
     }
 
+    public func verifySignature(httpClient: SwiftlyHTTPClient, archiveDownloadURL: URL, archive: URL) async throws {
+        guard (try? self.runProgram("gpg", "--version", quiet: true)) != nil else {
+            SwiftlyCore.print("gpg not installed, skipping signature verification.")
+            return
+        }
+
+        let foundKeys = (try? self.runProgram(
+            "gpg",
+            "--list-keys",
+            "swift-infrastructure@forums.swift.org",
+            "swift-infrastructure@swift.org",
+            quiet: true
+        )) != nil
+        guard foundKeys else {
+            SwiftlyCore.print("Swift PGP keys not imported, skipping signature verification.")
+            SwiftlyCore.print("To enable verification, import the keys from https://swift.org/keys/all-keys.asc")
+            return
+        }
+
+        SwiftlyCore.print("Refreshing Swift PGP keys...")
+        do {
+            try self.runProgram(
+                "gpg",
+                "--quiet",
+                "--keyserver",
+                "hkp://keyserver.ubuntu.com",
+                "--refresh-keys",
+                "Swift"
+            )
+        } catch {
+            throw Error(message: "Failed to refresh PGP keys: \(error)")
+        }
```

Comment:
This takes about 4 seconds to perform, so we could consider only running it every X days or something like that to avoid having to do that on every install. On the other hand, in the event of a key revocation, we'd want to refresh them immediately to ensure any invalid keys aren't ever used. What do you think would be a good policy here?